rest/src/main/java/fi/otavanopisto/pyramus/rest/StudentRESTService.java

```diff
@@ -1841,7 +1891,20 @@ public Response deleteStudentContactLogEntry(@PathParam("STUDENTID") Long studen
     if (!contactLogEntry.getStudent().getId().equals(contactLogEntry.getStudent().getId())) {
       return Response.status(Status.NOT_FOUND).build();
     }
-
+    
+    ContactLogAccess access = studentController.resolveContactLogAccess(student);
+    
+    if (access.equals(ContactLogAccess.NONE)) {
+      return Response.status(Status.FORBIDDEN).build();
+    }
+    
+    if (access.equals(ContactLogAccess.OWN)) {
+      if (contactLogEntry.getCreator() != null) {
```

Comment:
If creator is null, I would add `else { return Response.status(Status.FORBIDDEN).build(); }` so only admins have access to remove comments that have no creator